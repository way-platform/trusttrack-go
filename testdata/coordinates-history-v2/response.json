{
  "items": [
    {
      "object_id": "test-vehicle-001",
      "datetime": "2024-01-15T10:30:00.000Z",
      "trip_type": "BUSINESS",
      "ignition_status": "OFF",
      "position": {
        "latitude": 52.5200,
        "longitude": 13.4050,
        "altitude": 34,
        "direction": 90,
        "speed": 0,
        "satellites_count": 8
      },
      "inputs": {
        "calculated_inputs": {
          "fuel_consumption": 0.00,
          "fuel_level": 75.50,
          "mileage": 12500.25,
          "rpm": 0.00,
          "temperature": 18.5,
          "custom_input_1": 1.23,
          "custom_input_2": 2.34,
          "custom_input_3": 3.45,
          "custom_input_4": 4.56,
          "custom_input_5": 5.67,
          "custom_input_6": 6.78,
          "custom_input_7": 7.89,
          "custom_input_8": 8.90,
          "din1_working_time": 120.00,
          "din2_working_time": 240.00,
          "din3_working_time": 360.00,
          "din4_working_time": 480.00,
          "weight": 1750.00
        },
        "device_inputs": {
          "digital_input_1": false,
          "digital_input_2": true,
          "digital_input_3": false,
          "digital_input_4": false,
          "analog_input_1": 1.25,
          "analog_input_2": 2.50,
          "gsm_signal_strength": 85,
          "power_supply_voltage": 12.8,
          "battery_voltage": 4.2,
          "pcb_temperature": 25,
          "ibutton": "AB123456",
          "virtual_odometer": 12500.25,
          "input_trigger": 42,
          "priority": "LOW",
          "operator": 1001,
          "movement": "STILL",
          "hdop": "1.2",
          "din1_working_time_diff": 5,
          "din2_working_time_diff": 10,
          "din3_working_time_diff": 15,
          "din4_working_time_diff": 20,
          "virtual_odometer_diff": 0,
          "ecodrive_fuel_used_in_highest_gear": 0
        },
        "other": {
          "country_code_geonames": 276,
          "virtual_gps_odometer": 12501.00
        }
      }
    },
    {
      "object_id": "test-vehicle-001",
      "datetime": "2024-01-15T10:30:30.000Z",
      "trip_type": "BUSINESS",
      "ignition_status": "ON",
      "position": {
        "latitude": 52.5205,
        "longitude": 13.4055,
        "altitude": 35,
        "direction": 95,
        "speed": 15,
        "satellites_count": 10
      },
      "inputs": {
        "calculated_inputs": {
          "fuel_consumption": 2.1,
          "fuel_level": 75.45,
          "mileage": 12500.30,
          "rpm": 850.00,
          "temperature": 19.2,
          "custom_input_1": 1.24,
          "custom_input_2": 2.35,
          "custom_input_3": 3.46,
          "custom_input_4": 4.57,
          "custom_input_5": 5.68,
          "custom_input_6": 6.79,
          "custom_input_7": 7.90,
          "custom_input_8": 8.91,
          "din1_working_time": 125.00,
          "din2_working_time": 245.00,
          "din3_working_time": 365.00,
          "din4_working_time": 485.00,
          "weight": 1752.00
        },
        "device_inputs": {
          "digital_input_1": true,
          "digital_input_2": false,
          "digital_input_3": true,
          "digital_input_4": true,
          "analog_input_1": 1.28,
          "analog_input_2": 2.53,
          "gsm_signal_strength": 78,
          "power_supply_voltage": 13.2,
          "battery_voltage": 4.1,
          "pcb_temperature": 28,
          "ibutton": "AB123456",
          "virtual_odometer": 12500.30,
          "input_trigger": 15,
          "priority": "HIGH",
          "operator": 1001,
          "movement": "MOVING",
          "hdop": "0.8",
          "din1_working_time_diff": 5,
          "din2_working_time_diff": 10,
          "din3_working_time_diff": 15,
          "din4_working_time_diff": 25,
          "virtual_odometer_diff": 5,
          "ecodrive_fuel_used_in_highest_gear": 1
        },
        "other": {
          "country_code_geonames": 276,
          "virtual_gps_odometer": 12501.05
        }
      }
    },
    {
      "object_id": "test-vehicle-001",
      "datetime": "2024-01-15T10:31:00.000Z",
      "trip_type": "BUSINESS",
      "ignition_status": "ON",
      "position": {
        "latitude": 52.5210,
        "longitude": 13.4060,
        "altitude": 36,
        "direction": 100,
        "speed": 25,
        "satellites_count": 12
      },
      "inputs": {
        "calculated_inputs": {
          "fuel_consumption": 3.2,
          "fuel_level": 75.40,
          "mileage": 12500.50,
          "rpm": 1200.00,
          "temperature": 20.1,
          "custom_input_1": 1.25,
          "custom_input_2": 2.36,
          "custom_input_3": 3.47,
          "custom_input_4": 4.58,
          "custom_input_5": 5.69,
          "custom_input_6": 6.80,
          "custom_input_7": 7.91,
          "custom_input_8": 8.92,
          "din1_working_time": 130.00,
          "din2_working_time": 250.00,
          "din3_working_time": 370.00,
          "din4_working_time": 490.00,
          "weight": 1755.00
        },
        "device_inputs": {
          "digital_input_1": false,
          "digital_input_2": true,
          "digital_input_3": false,
          "digital_input_4": true,
          "analog_input_1": 1.30,
          "analog_input_2": 2.55,
          "gsm_signal_strength": 92,
          "power_supply_voltage": 13.5,
          "battery_voltage": 4.15,
          "pcb_temperature": 30,
          "ibutton": "AB123456",
          "virtual_odometer": 12500.50,
          "input_trigger": 88,
          "priority": "HIGH",
          "operator": 1001,
          "movement": "MOVING",
          "hdop": "0.6",
          "din1_working_time_diff": 5,
          "din2_working_time_diff": 10,
          "din3_working_time_diff": 15,
          "din4_working_time_diff": 30,
          "virtual_odometer_diff": 20,
          "ecodrive_fuel_used_in_highest_gear": 2
        },
        "other": {
          "country_code_geonames": 276,
          "virtual_gps_odometer": 12501.20
        }
      }
    },
    {
      "object_id": "test-vehicle-001",
      "datetime": "2024-01-15T10:32:30.000Z",
      "trip_type": "BUSINESS",
      "ignition_status": "ON",
      "position": {
        "latitude": 52.5215,
        "longitude": 13.4065,
        "altitude": 38,
        "direction": 105,
        "speed": 35,
        "satellites_count": 11
      },
      "inputs": {
        "calculated_inputs": {
          "fuel_consumption": 4.5,
          "fuel_level": 75.30,
          "mileage": 12501.00,
          "rpm": 1650.00,
          "temperature": 21.8,
          "custom_input_1": 1.26,
          "custom_input_2": 2.37,
          "custom_input_3": 3.48,
          "custom_input_4": 4.59,
          "custom_input_5": 5.70,
          "custom_input_6": 6.81,
          "custom_input_7": 7.92,
          "custom_input_8": 8.93,
          "din1_working_time": 135.00,
          "din2_working_time": 255.00,
          "din3_working_time": 375.00,
          "din4_working_time": 495.00,
          "weight": 1758.00
        },
        "device_inputs": {
          "digital_input_1": true,
          "digital_input_2": true,
          "digital_input_3": false,
          "digital_input_4": true,
          "analog_input_1": 1.35,
          "analog_input_2": 2.60,
          "gsm_signal_strength": 88,
          "power_supply_voltage": 13.8,
          "battery_voltage": 4.18,
          "pcb_temperature": 32,
          "ibutton": "AB123456",
          "virtual_odometer": 12501.00,
          "input_trigger": 99,
          "priority": "HIGH",
          "operator": 1001,
          "movement": "MOVING",
          "hdop": "0.5",
          "din1_working_time_diff": 10,
          "din2_working_time_diff": 15,
          "din3_working_time_diff": 20,
          "din4_working_time_diff": 90,
          "virtual_odometer_diff": 50,
          "ecodrive_fuel_used_in_highest_gear": 3
        },
        "other": {
          "country_code_geonames": 276,
          "virtual_gps_odometer": 12501.50
        }
      }
    },
    {
      "object_id": "test-vehicle-001",
      "datetime": "2024-01-15T10:35:00.000Z",
      "trip_type": "BUSINESS",
      "ignition_status": "ON",
      "position": {
        "latitude": 52.5220,
        "longitude": 13.4070,
        "altitude": 40,
        "direction": 110,
        "speed": 45,
        "satellites_count": 13
      },
      "inputs": {
        "calculated_inputs": {
          "fuel_consumption": 6.8,
          "fuel_level": 75.15,
          "mileage": 12502.00,
          "rpm": 2100.00,
          "temperature": 23.5,
          "custom_input_1": 1.27,
          "custom_input_2": 2.38,
          "custom_input_3": 3.49,
          "custom_input_4": 4.60,
          "custom_input_5": 5.71,
          "custom_input_6": 6.82,
          "custom_input_7": 7.93,
          "custom_input_8": 8.94,
          "din1_working_time": 140.00,
          "din2_working_time": 260.00,
          "din3_working_time": 380.00,
          "din4_working_time": 500.00,
          "weight": 1760.00
        },
        "device_inputs": {
          "digital_input_1": false,
          "digital_input_2": false,
          "digital_input_3": true,
          "digital_input_4": true,
          "analog_input_1": 1.40,
          "analog_input_2": 2.65,
          "gsm_signal_strength": 95,
          "power_supply_voltage": 14.1,
          "battery_voltage": 4.20,
          "pcb_temperature": 35,
          "ibutton": "AB123456",
          "virtual_odometer": 12502.00,
          "input_trigger": 75,
          "priority": "MEDIUM",
          "operator": 1001,
          "movement": "MOVING",
          "hdop": "0.4",
          "din1_working_time_diff": 15,
          "din2_working_time_diff": 20,
          "din3_working_time_diff": 25,
          "din4_working_time_diff": 150,
          "virtual_odometer_diff": 100,
          "ecodrive_fuel_used_in_highest_gear": 5
        },
        "other": {
          "country_code_geonames": 276,
          "virtual_gps_odometer": 12502.00
        }
      }
    }
  ],
  "continuation_token": "2024-01-15T11:00:00.000Z"
}
