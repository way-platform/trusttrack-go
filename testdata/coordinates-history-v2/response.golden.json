[
  {
    "objectId": "test-vehicle-001",
    "vehicleTime": "2024-01-15T10:30:00Z",
    "ignitionState": "OFF",
    "tripType": "BUSINESS",
    "position": {
      "satellitesCount": 8,
      "latitude": 52.52,
      "longitude": 13.405,
      "altitudeM": 34,
      "speedKmh": 0,
      "directionDeg": 90
    },
    "calculatedInputs": {
      "fuelConsumptionLifetimeL": 0,
      "fuelLevelPercent": 75.5,
      "odometerKm": 12500.25,
      "engineRpm": 0,
      "temperatureC": 18.5,
      "customInput1": 1.2300000190734863,
      "customInput2": 2.3399999141693115,
      "customInput3": 3.450000047683716,
      "customInput4": 4.559999942779541,
      "customInput5": 5.670000076293945,
      "customInput6": 6.78000020980835,
      "customInput7": 7.889999866485596,
      "customInput8": 8.899999618530273,
      "din1WorkingTime": 120,
      "din2WorkingTime": 240,
      "din3WorkingTime": 360,
      "din4WorkingTime": 480,
      "weightKg": 1750
    },
    "deviceInputs": {
      "gsmSignalStrength": 85,
      "operator": 1001,
      "ecodriveFuelUsedInHighestGear": 0,
      "ibutton": "AB123456",
      "digitalInput1": false,
      "digitalInput2": true,
      "digitalInput3": false,
      "digitalInput4": false,
      "analogInput1": 1.25,
      "analogInput2": 2.5,
      "din1WorkingTimeDiffS": 5,
      "din2WorkingTimeDiffS": 10,
      "din3WorkingTimeDiffS": 15,
      "din4WorkingTimeDiffS": 20,
      "powerSupplyVoltageV": 12.800000190734863,
      "batteryVoltageV": 4.199999809265137,
      "pcbTemperatureC": 25,
      "virtualOdometerKm": 12500.25,
      "inputTrigger": 42,
      "priority": "LOW",
      "movement": "STILL",
      "hdop": "1.2",
      "virtualOdometerDiff": 0
    },
    "other": {
      "countryCodeGeonames": 276,
      "virtualGpsOdometerKm": 12501
    }
  },
  {
    "objectId": "test-vehicle-001",
    "vehicleTime": "2024-01-15T10:30:30Z",
    "ignitionState": "ON",
    "tripType": "BUSINESS",
    "position": {
      "satellitesCount": 10,
      "latitude": 52.5205,
      "longitude": 13.4055,
      "altitudeM": 35,
      "speedKmh": 15,
      "directionDeg": 95
    },
    "calculatedInputs": {
      "fuelConsumptionLifetimeL": 2.0999999046325684,
      "fuelLevelPercent": 75.44999694824219,
      "odometerKm": 12500.2998046875,
      "engineRpm": 850,
      "temperatureC": 19.200000762939453,
      "customInput1": 1.2400000095367432,
      "customInput2": 2.3499999046325684,
      "customInput3": 3.4600000381469727,
      "customInput4": 4.570000171661377,
      "customInput5": 5.679999828338623,
      "customInput6": 6.789999961853027,
      "customInput7": 7.900000095367432,
      "customInput8": 8.90999984741211,
      "din1WorkingTime": 125,
      "din2WorkingTime": 245,
      "din3WorkingTime": 365,
      "din4WorkingTime": 485,
      "weightKg": 1752
    },
    "deviceInputs": {
      "gsmSignalStrength": 78,
      "operator": 1001,
      "ecodriveFuelUsedInHighestGear": 1,
      "ibutton": "AB123456",
      "digitalInput1": true,
      "digitalInput2": false,
      "digitalInput3": true,
      "digitalInput4": true,
      "analogInput1": 1.2799999713897705,
      "analogInput2": 2.5299999713897705,
      "din1WorkingTimeDiffS": 5,
      "din2WorkingTimeDiffS": 10,
      "din3WorkingTimeDiffS": 15,
      "din4WorkingTimeDiffS": 25,
      "powerSupplyVoltageV": 13.199999809265137,
      "batteryVoltageV": 4.099999904632568,
      "pcbTemperatureC": 28,
      "virtualOdometerKm": 12500.2998046875,
      "inputTrigger": 15,
      "priority": "HIGH",
      "movement": "MOVING",
      "hdop": "0.8",
      "virtualOdometerDiff": 5
    },
    "other": {
      "countryCodeGeonames": 276,
      "virtualGpsOdometerKm": 12501.0498046875
    }
  },
  {
    "objectId": "test-vehicle-001",
    "vehicleTime": "2024-01-15T10:31:00Z",
    "ignitionState": "ON",
    "tripType": "BUSINESS",
    "position": {
      "satellitesCount": 12,
      "latitude": 52.521,
      "longitude": 13.406,
      "altitudeM": 36,
      "speedKmh": 25,
      "directionDeg": 100
    },
    "calculatedInputs": {
      "fuelConsumptionLifetimeL": 3.200000047683716,
      "fuelLevelPercent": 75.4000015258789,
      "odometerKm": 12500.5,
      "engineRpm": 1200,
      "temperatureC": 20.100000381469727,
      "customInput1": 1.25,
      "customInput2": 2.359999895095825,
      "customInput3": 3.4700000286102295,
      "customInput4": 4.579999923706055,
      "customInput5": 5.690000057220459,
      "customInput6": 6.800000190734863,
      "customInput7": 7.909999847412109,
      "customInput8": 8.920000076293945,
      "din1WorkingTime": 130,
      "din2WorkingTime": 250,
      "din3WorkingTime": 370,
      "din4WorkingTime": 490,
      "weightKg": 1755
    },
    "deviceInputs": {
      "gsmSignalStrength": 92,
      "operator": 1001,
      "ecodriveFuelUsedInHighestGear": 2,
      "ibutton": "AB123456",
      "digitalInput1": false,
      "digitalInput2": true,
      "digitalInput3": false,
      "digitalInput4": true,
      "analogInput1": 1.2999999523162842,
      "analogInput2": 2.549999952316284,
      "din1WorkingTimeDiffS": 5,
      "din2WorkingTimeDiffS": 10,
      "din3WorkingTimeDiffS": 15,
      "din4WorkingTimeDiffS": 30,
      "powerSupplyVoltageV": 13.5,
      "batteryVoltageV": 4.150000095367432,
      "pcbTemperatureC": 30,
      "virtualOdometerKm": 12500.5,
      "inputTrigger": 88,
      "priority": "HIGH",
      "movement": "MOVING",
      "hdop": "0.6",
      "virtualOdometerDiff": 20
    },
    "other": {
      "countryCodeGeonames": 276,
      "virtualGpsOdometerKm": 12501.2001953125
    }
  },
  {
    "objectId": "test-vehicle-001",
    "vehicleTime": "2024-01-15T10:32:30Z",
    "ignitionState": "ON",
    "tripType": "BUSINESS",
    "position": {
      "satellitesCount": 11,
      "latitude": 52.5215,
      "longitude": 13.4065,
      "altitudeM": 38,
      "speedKmh": 35,
      "directionDeg": 105
    },
    "calculatedInputs": {
      "fuelConsumptionLifetimeL": 4.5,
      "fuelLevelPercent": 75.30000305175781,
      "odometerKm": 12501,
      "engineRpm": 1650,
      "temperatureC": 21.799999237060547,
      "customInput1": 1.2599999904632568,
      "customInput2": 2.369999885559082,
      "customInput3": 3.4800000190734863,
      "customInput4": 4.590000152587891,
      "customInput5": 5.699999809265137,
      "customInput6": 6.809999942779541,
      "customInput7": 7.920000076293945,
      "customInput8": 8.930000305175781,
      "din1WorkingTime": 135,
      "din2WorkingTime": 255,
      "din3WorkingTime": 375,
      "din4WorkingTime": 495,
      "weightKg": 1758
    },
    "deviceInputs": {
      "gsmSignalStrength": 88,
      "operator": 1001,
      "ecodriveFuelUsedInHighestGear": 3,
      "ibutton": "AB123456",
      "digitalInput1": true,
      "digitalInput2": true,
      "digitalInput3": false,
      "digitalInput4": true,
      "analogInput1": 1.350000023841858,
      "analogInput2": 2.5999999046325684,
      "din1WorkingTimeDiffS": 10,
      "din2WorkingTimeDiffS": 15,
      "din3WorkingTimeDiffS": 20,
      "din4WorkingTimeDiffS": 90,
      "powerSupplyVoltageV": 13.800000190734863,
      "batteryVoltageV": 4.179999828338623,
      "pcbTemperatureC": 32,
      "virtualOdometerKm": 12501,
      "inputTrigger": 99,
      "priority": "HIGH",
      "movement": "MOVING",
      "hdop": "0.5",
      "virtualOdometerDiff": 50
    },
    "other": {
      "countryCodeGeonames": 276,
      "virtualGpsOdometerKm": 12501.5
    }
  },
  {
    "objectId": "test-vehicle-001",
    "vehicleTime": "2024-01-15T10:35:00Z",
    "ignitionState": "ON",
    "tripType": "BUSINESS",
    "position": {
      "satellitesCount": 13,
      "latitude": 52.522,
      "longitude": 13.407,
      "altitudeM": 40,
      "speedKmh": 45,
      "directionDeg": 110
    },
    "calculatedInputs": {
      "fuelConsumptionLifetimeL": 6.800000190734863,
      "fuelLevelPercent": 75.1500015258789,
      "odometerKm": 12502,
      "engineRpm": 2100,
      "temperatureC": 23.5,
      "customInput1": 1.2699999809265137,
      "customInput2": 2.380000114440918,
      "customInput3": 3.490000009536743,
      "customInput4": 4.599999904632568,
      "customInput5": 5.710000038146973,
      "customInput6": 6.820000171661377,
      "customInput7": 7.929999828338623,
      "customInput8": 8.9399995803833,
      "din1WorkingTime": 140,
      "din2WorkingTime": 260,
      "din3WorkingTime": 380,
      "din4WorkingTime": 500,
      "weightKg": 1760
    },
    "deviceInputs": {
      "gsmSignalStrength": 95,
      "operator": 1001,
      "ecodriveFuelUsedInHighestGear": 5,
      "ibutton": "AB123456",
      "digitalInput1": false,
      "digitalInput2": false,
      "digitalInput3": true,
      "digitalInput4": true,
      "analogInput1": 1.399999976158142,
      "analogInput2": 2.6500000953674316,
      "din1WorkingTimeDiffS": 15,
      "din2WorkingTimeDiffS": 20,
      "din3WorkingTimeDiffS": 25,
      "din4WorkingTimeDiffS": 150,
      "powerSupplyVoltageV": 14.100000381469727,
      "batteryVoltageV": 4.199999809265137,
      "pcbTemperatureC": 35,
      "virtualOdometerKm": 12502,
      "inputTrigger": 75,
      "priority": "MEDIUM",
      "movement": "MOVING",
      "hdop": "0.4",
      "virtualOdometerDiff": 100
    },
    "other": {
      "countryCodeGeonames": 276,
      "virtualGpsOdometerKm": 12502
    }
  }
]